### PR TITLE
Fix #172

### LINKED ISSUE #172: ResponsiveContainer does not respond to resizing from large to small
ResponsiveContainer computes the size of the chart based on the current window size. It also seems to respond when you make the browser window larger, however if you make the browser window smaller, the chart stays large. Tested on chrome.

### PR DESCRIPTION
**Pull request by HHongSeungWoo**
## Description

I have fixed issue #172

## Related Issue

#172

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] I have added a storybook story or extended an existing story to show my changes
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by ckifer** [comment]
Can you explain more of why this fixes the issue?

**Comment by ckifer** [comment]
Pulled this down - this is definitely a breaking change. Try running the demo project and looking at `ResponsiveContainer`
image

Height gets quite messed up (though everything is much much faster)

**Comment by HHongSeungWoo** [comment]
@ckifer I wanted to make non-breaking changes. I've made the changes again.

**Comment by HHongSeungWoo** [comment]
The test is failing. Please wait for a moment

**Comment by HHongSeungWoo** [comment]
> Can you explain more of why this fixes the issue?

I've fixed the issue where the flexbox wasn't shrinking due to fixed inner item length

I'm not good at English, so I can't provide a long explanation. :smile:  :sob:

**Comment by ckifer** [comment]
Okay, it's late here so I'll take a look tomorrow. 

And you are good at English!

File: src/component/ResponsiveContainer.tsx [hunk 1]
```diff
@@ -162,6 +164,7 @@ export const ResponsiveContainer = forwardRef<HTMLDivElement, Props>(
         style={{ ...style, width, height, minWidth, minHeight, maxHeight }}
         ref={containerRef}
       >
+        <style>{`.recharts-wrapper > .recharts-surface {width: 100%; height: 100%;}`}</style>
```
Comment (ckifer) [inline src/component/ResponsiveContainer.tsx:167]: I don't think its a good idea to have a style tag in the body of the document. Is there any way we can do this another way?

**Review by ckifer — CHANGES_REQUESTED** [review]

File: src/component/ResponsiveContainer.tsx (hunk 1 above)
Comment (HHongSeungWoo) [reply src/component/ResponsiveContainer.tsx]: I removed the style tags and added inline styles.

**Review by nikolasrieble — APPROVED** [review]
I tested in both demo and storybook, and could not find any issue. This is wonderful! Thank you!

**Review by ckifer — APPROVED** [review]

**Comment by ckifer** [comment]
@HHongSeungWoo https://www.chromatic.com/test?appId=63da8268a0da9970db6992aa&id=653f5f167ae6c3898ab13ce7

That diff is concerning and is introduced by this PR. Why would we have one `tspan` versus one for every word with this change?

**Comment by HHongSeungWoo** [comment]
Ah... I hadn't thought of wrapping individual components in a container.
the simplest solution would be to pass styles only to the chart.

**Comment by HHongSeungWoo** [comment]
We feel that we are allowing too much, which results in terrible side effects. 
We believe that there should be appropriate constraints in the next major version :cry:

**Comment by HHongSeungWoo** [comment]
https://www.chromatic.com/test?appId=63da8268a0da9970db6992aa&id=653fe597bee4c9fdc1a8f6af
the original ResponsiveContainer had center alignment as the default
do you think this should also be maintained?

**Comment by ckifer** [comment]
+1, completely agree 😓. This is inherited legacy as I've said before, if I could restart from the ground up I would not allow the "features" that cause issues like this like implicit props and prop spreading

**Comment by ckifer** [comment]
>the original ResponsiveContainer had center alignment as the default
do you think this should also be maintained?

imo it should be maintained for now yes as changing it would be breaking. If there are issues with center alignment we can change that in v3+

**Comment by HHongSeungWoo** [comment]
Ok~! I'll make the changes and PR when ready

**Pull request merged**